spring-cloud-starter-task-composedtaskrunner/src/main/java/org/springframework/cloud/task/app/composedtaskrunner/TaskLauncherTasklet.java

```diff
@@ -96,44 +102,40 @@ public void setArguments(List<String> arguments) {
 	/**
 	 * Executes the task as specified by the taskName with the associated
 	 * properties and arguments.
+	 *
 	 * @param contribution mutable state to be passed back to update the current step execution
 	 * @param chunkContext contains the task-execution-id used by the listener.
 	 * @return Repeat status of FINISHED.
 	 */
 	@Override
 	public RepeatStatus execute(StepContribution contribution,
-			ChunkContext chunkContext) throws Exception {
-		String tmpTaskName = this.taskName.substring(0,
-				this.taskName.lastIndexOf('_'));
+			ChunkContext chunkContext) {
+		if (this.executionId == null) {
+			this.timeout = System.currentTimeMillis() +
+					this.composedTaskProperties.getMaxWaitTime();
+			logger.debug("Wait time for this task to complete is " +
+					this.composedTaskProperties.getMaxWaitTime());
+			logger.debug("Interval check time for this task to complete is " +
+					this.composedTaskProperties.getIntervalTimeBetweenChecks());
+
+			String tmpTaskName = this.taskName.substring(0,
+					this.taskName.lastIndexOf('_'));
+
+			List<String> args = this.arguments;
+
+			ExecutionContext stepExecutionContext = chunkContext.getStepContext().getStepExecution().
+					getExecutionContext();
+			if (stepExecutionContext.containsKey("task-arguments")) {
+				args = (List<String>) stepExecutionContext.get("task-arguments");
+			}
```

Comment:
Is this `if` statement for restart?  If so, we should use Spring Batch's `StepScope` and property injection instead of this kind of manually digging these values out of the context.  If not, what is this used for?